dataflow/src/main/java/org/checkerframework/dataflow/cfg/builder/CFGTranslationPhaseOne.java

```diff
@@ -3469,9 +3469,17 @@ public Node visitNewArray(NewArrayTree tree, Void p) {
     public Node visitNewClass(NewClassTree tree, Void p) {
         // see JLS 15.9
 
+        DeclaredType classType = (DeclaredType) TreeUtils.typeOf(tree);
+        TypeMirror enclosingClassType = classType.getEnclosingType();
         Tree enclosingExpr = tree.getEnclosingExpression();
+        Node enclosingExprNode;
         if (enclosingExpr != null) {
-            scan(enclosingExpr, p);
+            enclosingExprNode = scan(enclosingExpr, p);
+        } else if (enclosingClassType.getKind() == TypeKind.DECLARED) {
```

Comment:
What other options are there? Maybe variable `enclosingClassType` shouldn't contain `Class` if it's not guaranteed to be a class?

Reply:
Do you mean I need to change the variable name? That makes sense.